job/job.go

```diff
@@ -9,9 +10,17 @@ import (
 	"github.com/tinkerbell/boots/conf"
 	"github.com/tinkerbell/boots/dhcp"
 	"github.com/tinkerbell/boots/packet"
+	tw "github.com/tinkerbell/tink/protos/workflow"
 )
 
-var client *packet.Client
+var (
+	client    *packet.Client
+	connectTo endpointClient
+)
+
+func init() {
+	connectTo = &connect{}
+}
```

Comment:
This is still a global var. And also now its a new "large" interface, when the just a func was better imo. Why not just make `isHardwareActiveWorkflow` take in a function?

Reply:
I have modified the method to pass `connectTo` in method. 

Earlier, it was single method so passing the method as parameter could have been fine. But as the complexity increases, number of methods to be mocked also increases, interface would be eventually a better option for testing.